### PR TITLE
Adjust probability of n_sources at inference time

### LINKED ISSUE #509: Adjust probability of n_sources in detection encoder at test time
By adjusting the output of the detection encoder  

```
beta_0^{\star} = beta_0 + log (pi / (1 - pi)) - log ( pi^{\tilde}  / (1 - pi^{\tilde}) 
```

used for logistic regression but should work with softmax too. `pi` is correct prior (test data) and `pi^{\tilde}` is the training set prior.

### PR DESCRIPTION
**Pull request by dereklhansen**
Closes #509.

### CODE REVIEW

**Comment by dereklhansen** [comment]
```
output-adj-mean/semisynthetic_figures
detections
           recall  expected_recall  precision  expected_precision    n  n_galaxies  n_galaxies_detected
18       0.942857              NaN   1.000000                 NaN   35        13.0                   13
19       0.859375              NaN   1.000000                 NaN   64        37.0                   30
20       0.820513              NaN   1.000000                 NaN   78        52.0                   42
21       0.721804              NaN   1.000000                 NaN  133        89.0                   53
22       0.470852              NaN   0.906832                 NaN  223       142.0                   46
23       0.236994              NaN   0.928571                 NaN  173       107.0                   13
overall  0.600000         0.795509   0.971554            0.896567  740       453.0                  218
accuracy
         class_acc  expected_accuracy  galaxy_accuracy  expected_galaxy_accuracy  star_accuracy  expected_star_accuracy    n  acc_n_matches  n_galaxies
18        1.000000           0.999623         1.000000                  0.999456       1.000000                0.999725   35             33        13.0
19        1.000000           0.999057         1.000000                  0.998247       1.000000                0.999802   64             55        37.0
20        1.000000           0.996692         1.000000                  0.995689       1.000000                0.998548   78             64        52.0
21        0.968750           0.980017         0.962264                  0.967588       0.976744                0.994931  133             96        89.0
22        0.904762           0.862934         0.847826                  0.836302       0.949153                0.899357  223            105       142.0
23        0.707317           0.792027         0.769231                  0.849222       0.678571                0.774154  173             41       107.0
overall   0.941441           0.928191         0.944954                  0.920595       0.938053                0.936092  740            444       453.0

output-adj-mean/sdss_figures
detections
           recall  expected_recall  precision  expected_precision    n  n_galaxies  n_galaxies_detected
18       0.937500              NaN   1.000000                 NaN   32          10                    8
19       0.750000              NaN   0.939394                 NaN   56          29                   19
20       0.682353              NaN   0.888889                 NaN   85          59                   37
21       0.698864              NaN   0.800000                 NaN  176         132                   81
22       0.380753              NaN   0.766667                 NaN  239         158                   33
23       0.292035              NaN   0.593750                 NaN  113          47                    3
overall  0.576132          0.68407   0.815534            0.863769  729         442                  201
accuracy
         class_acc  expected_accuracy  galaxy_accuracy  expected_galaxy_accuracy  star_accuracy  expected_star_accuracy    n  acc_n_matches  n_galaxies
18        1.000000           0.994669         1.000000                  0.997921       1.000000                0.993530   32             30          10
19        1.000000           0.997537         1.000000                  0.991329       1.000000                0.999865   56             42          29
20        0.965517           0.998613         0.945946                  0.998104       1.000000                0.999162   85             58          59
21        0.845528           0.939680         0.777778                  0.917994       0.976190                0.965433  176            123         132
22        0.758242           0.844040         0.575758                  0.819353       0.862069                0.888478  239             91         158
23        0.545455           0.755813         1.000000                  0.756554       0.500000                0.755659  113             33          47
overall   0.857143           0.893174         0.820895                  0.870910       0.890411                0.913694  729            420         442

output-adj-mean/simulated_figures
detections
           recall  expected_recall  precision  expected_precision     n  n_galaxies  n_galaxies_detected
18       0.985782              NaN   0.980296                 NaN   211       144.0                  141
19       0.953079              NaN   0.959119                 NaN   341       235.0                  221
20       0.897036              NaN   0.936047                 NaN   641       436.0                  378
21       0.821285              NaN   0.835546                 NaN   996       702.0                  535
22       0.567453              NaN   0.701953                 NaN  1653      1183.0                  533
23       0.216309              NaN   0.750751                 NaN  1165       862.0                   79
overall  0.641404         0.709591   0.808660            0.894276  5212      3682.0                 2025
accuracy
         class_acc  expected_accuracy  galaxy_accuracy  expected_galaxy_accuracy  star_accuracy  expected_star_accuracy     n  acc_n_matches  n_galaxies
18        0.995192           0.995038         0.992908                  0.992805       1.000000                0.999375   211            208       144.0
19        0.993846           0.983172         1.000000                  0.978073       0.980769                0.994789   341            325       235.0
20        0.986087           0.982467         0.984127                  0.975060       0.989848                0.996177   641            575       436.0
21        0.935208           0.930135         0.921495                  0.917477       0.961131                0.961734   996            818       702.0
22        0.857143           0.867798         0.840525                  0.862558       0.879012                0.880639  1653            938      1183.0
23        0.666667           0.764055         0.822785                  0.879692       0.595376                0.736009  1165            252       862.0
overall   0.911158           0.909805         0.926420                  0.912873       0.887709                0.904029  5212           3343      3682.0
```

**Review by jeff-regier — APPROVED** [review]
Fast work. Looks great
